public/app/plugins/datasource/prometheus/datasource.tsx

```diff
@@ -797,7 +797,7 @@ export class PrometheusDatasource
         timeValueTuple.push([timeStampValue, valueValue]);
       });
 
-      const activeValues = timeValueTuple.filter((value) => value[1] >= 1);
+      const activeValues = timeValueTuple.filter((value) => value[1] > 0);
```

Comment:
@ivanahuckova @torkelo Is there any historical context for why we are filtering values greater then 0, or should this be fixed?

Reply:
no idea

Reply:
I did line history check and it seems that it was introduced in https://github.com/grafana/grafana/pull/22225 that fixed https://github.com/grafana/grafana/issues/18174.

Reply:
Thanks @ivanahuckova! It looks as though that issue was to prevent a bunch of zero values from being returned as they can slow things down. When testing similar queries on this branch vs main I'm not seeing any difference in behavior. 

@secustor it looks like you were the author of this change, can you think of any problems that might be introduced if we change this to allow values greater than 0?

Reply:
> @secustor it looks like you were the author of this change, can you think of any problems that might be introduced if we change this to allow values greater than 0?     

Nothing from top of my head. 

That said, the screenshots with fix look wrong. Is the step size 15s and datapoint interval 30s in the example? In that case logic would still be working correctly, else it should be a region annotation

Reply:
Yeah I might have messed up those screenshots, let me try again using a consistent step size:

Query:
<img width="1193" alt="image" src="https://user-images.githubusercontent.com/109082771/190664419-a80bc8b0-ea26-4434-99da-eabfe0c46f8c.png">

Annotation 1:
<img width="1470" alt="image" src="https://user-images.githubusercontent.com/109082771/190664595-4d4bb4f7-0793-49e4-9252-e2e9dbb9d423.png">

Annotation 2:
<img width="1471" alt="image" src="https://user-images.githubusercontent.com/109082771/190664712-11c12ca8-ab03-4f72-9d25-69855b65cd2f.png">

I'm still not sure I'm fully understanding here, @secustor does this help clarify?

Reply:
LGTM